.github/workflows/reusable-build-test-release.yml

```diff
@@ -2377,8 +2377,9 @@ jobs:
 
   publish:
     if: |
-      (!cancelled() && needs.pre-publish.result == 'success' && github.event_name != 'pull_request' && github.event_name != 'schedule') ||
+      (!cancelled() && needs.pre-publish.result == 'success' && github.event_name == 'push' && github.event.inputs.custom-version == '') ||
```

Comment:
This needs to be verified, but could we simplify the condition to be just: `!cancelled() && needs.pre-publish.result == 'success' && (github.event_name == 'push' || needs.validate-custom-version.result == 'success')`